e2e/playground.py

```diff
@@ -0,0 +1,2135 @@
+    # separator: str = " to ",
+    # width: Optional[str] = None,
+    # autoclose: bool = True,
+
+    loc_separator: Locator
+    loc_start: Locator
+    loc_end: Locator
+    date_start: _DateBase
+    date_end: _DateBase
+
+    def __init__(self, page: Page, id: str) -> None:
+        super().__init__(
+            page,
+            id=id,
+            loc="input[type=text].form-control",
+            loc_label=f"label#{id}-label",
+            loc_container=f"div#{id}.shiny-input-container",
+        )
+        self.loc_separator = self.loc_container.locator(".input-daterange > span")
+        self.loc_start = self.loc.nth(0)
+        self.loc_end = self.loc.nth(1)
+        self.date_start = _DateBase(
+            page,
+            id=id,
+            loc=self.loc_start,
+            loc_label=self.loc_label,
+            loc_container=self.loc_container,
+        )
+        self.date_end = _DateBase(
+            page,
+            id=id,
+            loc=self.loc_end,
+            loc_label=self.loc_label,
+            loc_container=self.loc_container,
+        )
+
+    def set(
+        self,
+        # TODO-barret; Should this be a list or a tuple?
+        value: typing.Union[
+            typing.Tuple[
+                typing.Union[datetime.date, str, None],
+                typing.Union[datetime.date, str, None],
+            ],
+            None,
+        ],
+        *,
+        timeout: Timeout = None,
+    ) -> None:
+        if value is None:
+            value = (None, None)
+        self.date_start.set(value=value[0], timeout=timeout)
+        self.date_end.set(value=value[1], timeout=timeout)
+
+    def expect_value(
+        self,
+        # TODO-barret; Should this be a list or a tuple?
+        value: typing.Union[
+            typing.Tuple[
+                typing.Union[datetime.date, PatternOrStr, None],
+                typing.Union[datetime.date, PatternOrStr, None],
+            ],
+            None,
+        ],
+        *,
+        timeout: Timeout = None,
+    ) -> None:
+
+        start_val = None
+        end_val = None
+        if not (value is None):
+            # Not using `_date_str()` as we want ability to supply non ISO formatted dates
+            start_val = value[0]
+            if isinstance(start_val, datetime.date):
+                start_val = str(start_val)
+            end_val = value[1]
+            if isinstance(end_val, datetime.date):
+                end_val = str(end_val)
+
+        if start_val is None:
+            start_val = ""
+        if end_val is None:
+            end_val = ""
+
+        # We can not use `[value={value}]` within Locators.
+        # The physical `value` attribute is never set, so we can not select on it.
+        # We must as the start and end values individually, rather than at the same time like the checkboxgroup input.
+        self.date_start.expect_value(start_val, timeout=timeout)
+        self.date_end.expect_value(end_val, timeout=timeout)
```

Comment:
Add note that these two lines should be atomic (when we do the project to have composable assertions)